### ISSUE TITLE
[discussion] About how (often) to run integration tests against staging.

### ISSUE THREAD

**Issue by geropl** [gitpod-io/gitpod#4651]
This issue is meant to continue the discussion started on [this PR](https://github.com/gitpod-com/gitpod/pull/5562).

The PR enables sequential execution of integration tests against all staging clusters **in sync** from inside the deployment job.

Pro:
 - easy to give feedback to Slack/deployment job
 - synchronized (bc deployment job is already synchronized bc it's running as a `CronJob`)
 Cons:
  - each test execution might take up to 10-15mins (times two clusters), adding to the deployment jobs time
 
 **Alternative: async execution**
 We could have a separate `CronJob` which regularly (every 5mins) checks for a new version, and if there is one, executes the integration test.
 Pros:
  - parallel execution in US/EU (one `CronJob` per cluster)
  - contains code/config for feedback channel (Slack)
  - ability two use other triggers as well (execute after certain time without new version, say every 24 hours e.g.)
  - synchronized bc it's a `CronJob`
  - we could re-use this job and deploy it to prod (Self-Hosted?) as well :bulb: 
 Cons:
  - we're not guaranteed to test _every_ version on staging because of polling/delays in integration tests execution (lost update)
  
  
  /cc @ArthurSens @mads-hartmann

**Comment by meysholdt** [comment]
Proposal: run tests in sync in Werft, but potentially in parallel. Advantages:
* keep Werft as the "UI" that show the tests log output / results / failures.
* no performance penalty due to parallelisation
* fastest-possible feedback for developers, because (1)  tests run immediately after (or as part of) deploy-to-staging and (2) every version is tested.
* we already have a CronJob that monitors for new versions and we can keep using it.

Potential con:
* this is too expensive (Walrus?)

Two options when new changes come in while the tests run:
* not an option: deploy-to-staging while the tests run. This is not an option because the test-result would be meaningless.
* option1: postpone deploy-to-staging until the tests have finished.
* option2: abort the tests, deploy-to-staging, run the tests on the new version

self-hosted: While I see the value of also running the tests against self-hosted, I don't atm see the value of packaging the logic to invoke the tests as part of GP self-hosted.
